pennylane/ops/op_math/adjoint_class.py

```diff
@@ -204,26 +204,36 @@ def __init__(self, base=None, do_queue=True, id=None):
     def __repr__(self):
         return f"Adjoint({self.base})"
 
+    # pylint: disable=protected-access
+    def _check_batching(self, params):
+        self.base._check_batching(params)
```

Comment:
As `SymbolicOp.__init__` does not inherit from `Operator.__init__`, `Adjoint._check_batching` is not being called anywhere, as far as I am aware. Maybe I am missing something here?
My suggestion would be to leave the definitions of `batch_size` and `ndim_params` from below in, but simply skip this one.

Reply:
Oh I see, I was indeed missing something.

I'm definitely in favour of implementing a property setter. I think the operator algorithmic work is really highlighting edge cases in the sense that it makes use of class features standard ops don't use, and this should definitely be made clearer!

I'm wondering: Are you thinking of an approach that would make it unnecessary for developers to think about the `batch_size` updating at all? :thinking:

Reply:
It'd be great if someone could just do:
```
op.data = new_data
```
and have batch size automatically updated.  But we can discuss this later elsewhere.